### PR TITLE
Redux

### PR DESCRIPTION
**Pull request by victorokoroji**
_In these project,_ 

- Created redux store
- Add actions for books and categories
- dispatch actions to store

### CODE REVIEW

File: src/redux/configureStore.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { createStore, combineReducers } from 'redux';
+import bookReducer from './books/books';
+import categoryReducer from './categories/categories';
```
Comment (Cathella) [inline src/redux/configureStore.js]: - [ ] Please rename this to `categoriesReducer` as defined in the referenced `categories.js`

File: src/redux/books/books.js [hunk 2]
```diff
@@ -1,77 +1,36 @@
...
     case REMOVE_BOOK:
-      return [...state].filter((book) => action.id !== book.id);
-    case UPDATE_PROGRESS:
-      return [...state].map((book) => action.id === book.id) ? action.book : book;
-    case ADD_COMMENT:
-      return [...state].map((book) => action.id === book.id) ? action.book : book;
+      return {
+        ...state,
+        books: payload,
+      };
```
Comment (Cathella) [inline src/redux/books/books.js]: - [ ] HINT: For the remove book action you can use filter by the id key to make sure that a book is removed from an array - i.e.: 
```
   return state.filter(book => book.id !== id);
```

**Review by Cathella — CHANGES_REQUESTED** [review]
## 🟡 Status: Changes Required 🛑

### Hi @vickymarz 👋🏻, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## To Highlight 👏🏻

✅ No linter errors.
✅ You used correct Github Flow.
✅ Great use of the ducks pattern.


## Required Changes ♻️
_Check the comments under the review._

- [ ] When I load the project on a local server, I see a series of console errors land a blank page. Please fix the console errors.

image


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

i-guess-we-have-to-give-it-another-shot-liz-lemon


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 2 above)
Comment (victorokoroji) [reply src/redux/books/books.js]: This is great. I will update this. Thanks

File: src/redux/configureStore.js (hunk 1 above)
Comment (victorokoroji) [reply src/redux/configureStore.js]: Thanks for spotting this.

**Review by BrianSammit — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hello @vickymarz, 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

TO HIGHLIGHT:

- Your project is professional 
- Your repo is professional 
- Well structured files 


Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**